step/src/main/java/com/google/sps/servlets/PrefServlet.java

```diff
@@ -62,12 +63,11 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       response.sendRedirect("index.jsp");
     }
 
-    entity.setProperty("q1", q1);
-    entity.setProperty("q2", q2);
-    entity.setProperty("q3", q3);
-    entity.setProperty("q4", q4);
-    entity.setProperty("q5", q5);
-    datastore.put(entity);
+    // Ensuring that only users who have filled all of the info fields out are added to the database
+    if (q1 != null && q2 != null && q3 != null && q4 != null &&
+      q5 != null) {
+      DatabaseHandler.addUserPref(id, q1, q2, q3, q4, q5);
+    }
```

Comment:
Again, I think users should have the right to omit answers and not provide data if they choose to do so. It should be our job to accommodate to their comfort levels, not their job to be convenient for us. I'm fine if users have to answer all questions in this PR, but I think this should only be temporary

Reply:
Yes I agree with you! However, currently we have a limited number of users and a limited number of questions. Therefore, in order to facilitate the matching process, it's better to keep 2 choice questions. Beside, I agree that all the questions should not be mandatory. However, if the users just don't answer 3 questions, there would be no way to match them because our minimum questions threshold are 3 and we don't have any alternative questions.